src/org/zaproxy/zap/extension/ascanrulesBeta/UsernameEnumeration.java

```diff
@@ -160,15 +166,18 @@ public void scan() {
 						// fragment, and POST params
 						// are possibly different from the login page.
 						loginUrl = true;
-						break;
+						log.info(requestUri.toString()+" falls within a context, and is the defined Login URL. Scanning for possible Username Enumeration vulnerability.");
+						break;//Stop checking
 					}
 				}
 			}
 						
 			//the Username Enumeration scanner will only run for logon pages
 			if (loginUrl == false) {
-				log.info("For the Username Enumeration scanner to actually scan this URL, the URL *must* be added to a context, and flagged as the login request in that context!");
-				return;
+				if(log.isDebugEnabled()) {
+					log.debug(requestUri.toString()+" falls within a context, but is not the defined Login URL.");
```

Comment:
The "falls within a context" should be changed, we are not checking that the URLs are in a context.